src/observer.tsx

```diff
@@ -1,31 +1,31 @@
-/* eslint-disable react/display-name */
-import React, { Component, forwardRef, memo } from "react"
+import * as React from "react"
 import { observer as observerLite, Observer } from "mobx-react-lite"
 
 import { makeClassComponentObserver } from "./observerClass"
+import { IReactComponent } from "./index"
 
-const hasSymbol = typeof Symbol === "function" && Symbol.for
+const hasSymbol: false | ((key: string) => symbol) = typeof Symbol === "function" && Symbol.for
 
 // Using react-is had some issues (and operates on elements, not on types), see #608 / #609
 const ReactForwardRefSymbol = hasSymbol
     ? Symbol.for("react.forward_ref")
-    : typeof forwardRef === "function" && forwardRef(() => {})["$$typeof"]
+    : typeof React.forwardRef === "function" && React.forwardRef((props: any) => null)["$$typeof"]
 
 const ReactMemoSymbol = hasSymbol
     ? Symbol.for("react.memo")
-    : typeof memo === "function" && memo(() => {})["$$typeof"]
+    : typeof React.memo === "function" && React.memo((props: any) => null)["$$typeof"]
 
 /**
  * Observer function / decorator
  */
-export function observer(componentClass) {
-    if (componentClass.isMobxInjector === true) {
+export function observer(component: IReactComponent<any>): IReactComponent<any> {
+    if (component["isMobxInjector"] === true) {
```

Comment:
note that quoting doesn't make any difference to typescript, so there is no need to do this. If you want to access some property TS doesn't know about (which will happen regularly in a complicated meta programming lib like this, the cleaner fix imho is to put a `// @ts-ignore` in the line above it

Reply:
It does difference, without quoting it will complain about that property missing. I would argue it's cleaner to add ignore, adds extra noise while with quotes it just works directly.

Reply:
that sounds like the tsconfig is off, as in normal strict settings the type inference for both of them should be exactly the same since `"isMobXInjector"` is a constant expression: http://www.typescriptlang.org/play/#code/MYewdgzgLgBAHjAvDA3gKBpmBDAXDARjQF801RIQAbAUwDoqQBzACjjuwCMBKAbnPARq9RqzgBtAERdJAXT5A

Reply:
It did complain because of 'strict' and so I added 'suppressImplicitAnyIndexErrors' to the tsconfig. I forgot to revert the use of string index on others.  Thanks for pointing that out.  I try not to make use of ts-ignore unless I have too.

Reply:
Actually my comment about making use of `suppressImplicityAnyIndexErrors` isn't applicable in this specific situation, I don't think, because we're being explicit. The compiler does still complain, like Daniel said.  Also, I don't believe the config is "off", since `yarn run tsc -w` should be using the closest configuration in the project.  Ultimately, it should work just the same and shouldn't cause any regressions, which is really what I'm aiming for in this PR.